Library/Homebrew/list.sh

```diff
@@ -0,0 +1,79 @@
+# does the quickest output of brew list possible for no named arguments.
+# HOMEBREW_CELLAR, HOMEBREW_PREFIX are set by brew.sh
+# shellcheck disable=SC2154
+homebrew-list() {
+  case "$1" in
+    # check we actually have list and not e.g. listsomething
+    list) ;;
+    list*) return 1 ;;
+    *) ;;
+  esac
+
+  local ls_args=()
+  local formula=""
+  local cask=""
+
+  for arg in "$@"
+  do
+    case "${arg}" in
+      # check for flags passed to ls
+      -1 | -l | -r | -t) ls_args+=("${arg}") ;;
+      --formula | --formulae) formula=1 ;;
+      --cask | --casks) cask=1 ;;
+      # reject all other flags
+      -* | *) return 1 ;;
+    esac
+  done
+
+  echo "list2 $*"
```

Comment:
Maybe we don't want to keep this?

Reply:
@carlocab This is essential functionality!

(😅)